README.md

```diff
@@ -1,56 +1,71 @@
+
+The goal of the Cosmos-SDK is to allow developers to easily create their custom interoperable blockchain applications within the Cosmos Network without having to code every single functionality of their application, thus abstracting away the complexities of building a Tendermint ABCI application. We envision the SDK as the `npm`-like framework to build secure blockchain applications on top of Tendermint.
+
+In terms of its design, the SDK optimizes flexibility and security. The framework is designed around a modular execution stack which allows applications to mix and match elements as desired. In addition, all modules are sandboxed for greater application security.
+
+It is based on two major principles:
+
+- **Composability**: Anyone can create a module for the Cosmos-SDK, and using the already-built modules in your blockchain is as simple as importing them into your application.  As a developer, you only have to create the modules required by your application that do not already exist.
+
+- **Capabilities**: The SDK is inspired by capabilities-based security, and informed by years of wrestling with blockchain state machines. Most developers will need to access other modules when building their own modules. Given that the Cosmos-SDK is an open framework, we designed it using object-capabilities (_ocaps_) based principles. This is because we assume that some modules are malicious. In practice, this means that instead of having each module keep an access control list to give access to other modules, each module implements `keepers` that can be passed to other modules to grant a pre-defined set of capabilities. For example, if an instance of module A's `keepers` is passed to module B, the latter will be able to call a restricted set of module A's functions. The capabilities of each `keeper` are defined by the module's developer, and it's the job of the application developer to instantiate and pass a `keeper` from module to module properly. For a deeper look at capabilities, you can read this [article](http://habitatchronicles.com/2017/05/what-are-capabilities/).
+
+_Note: For now the Cosmos-SDK only exists in [Golang](https://golang.org/), which means that developers can only develop SDK modules in Golang. In the future, we expect that the SDK will be implemented in other programming languages. The Cosmos team will also financially support their development._
+
+## Application architecture
+
+#### Modules
+
+A `module` is a fundamental unit in the Cosmos-SDK. Each module is an extension to the BaseApp functionalities that defines its own transactions, handles its own state as well as the state transition logic. The Cosmos-SDK has all the necessary pre-built modules to add functionality on top of a `BaseApp`, which is the template to build a blockchain dApp in Cosmos. Some of the most important ones are:

```

Comment:
"extension to the" is odd, I think more correct phrase is "extension of the" .... ALSO what is BaseApp - the reader has no idea, this concept should first be introduced before talking about how the developer extends it. .......... -> later on you say: 
>top of a `BaseApp`, which is the template to build a blockchain dApp in Cosmos

  ^ maybe that idea should come earlier in the paragraph

Reply:
`defines its own transactions` -> `defines transactions`

Reply:
`handles its own state as well as the state transition logic.` -> `handles application state and state transition logic.`